devicetypes/smartthings/eaton-accesory-dimmer.src/eaton-accessory-dimmer.groovy

```diff
@@ -0,0 +1,286 @@
+	handleDimmerValue(cmd.value)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd){
+	def delayedOff = settings.delayedOffTime ? settings.delayedOffTime : 10
+	def dimmerRamp = settings.dimmerRampTime ? settings.dimmerRampTime : 5
+	//settings are read-only, so send event if settings are out of sync
+	if(cmd.parameterNumber == 1 && cmd.configurationValue[0] != delayedOff) {
+		createEvent([descriptionText: "$device.displayName delayed off time settings are out of sync. Please save device preferences.",
+				isStateChange: true])
+	} else if (cmd.parameterNumber == 5 && cmd.configurationValue[0] != getPowerUpStateValue()) {
+		createEvent([descriptionText: "$device.displayName power up state settings are out of sync. Please save device preferences.",
+				isStateChange: true])
+	} else if (cmd.parameterNumber == 7 && cmd.configurationValue[0] != dimmerRamp){
+		createEvent([descriptionText: "$device.displayName dimmer ramp time settings are out of sync. Please save device preferences.",
+				isStateChange: true])
+	}else {
+		return null
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	createEvent(descriptionText: "$device.displayName: $cmd", isStateChange: true)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+	log.debug "ManufacturerSpecificReport: ${cmd}"
+	if (cmd.manufacturerName) {
+		updateDataValue("manufacturer", cmd.manufacturerName)
+	}
+	if (cmd.productTypeId) {
+		updateDataValue("productTypeId", cmd.productTypeId.toString())
+	}
+	if (cmd.productId != null) {
+		updateDataValue("productId", cmd.productId.toString())
+	}
+	return null
+}
+
+private handleDimmerValue(newValue){
+	def result = []
+	result << createEvent(name: "switch", value: newValue > 0 ? "on" : "off")
+	result << createEvent(name: "level", value: newValue)
+	return result
+}
+
+private handleOnOff(newValue){
+	def cmds = []
+	cmds << new physicalgraph.device.HubAction(zwave.basicV1.basicSet(value: newValue).format())
+	cmds << new physicalgraph.device.HubAction(zwave.basicV1.basicGet().format())
+	sendDelayedCommandsAccordingToDimmerRampTime(cmds)
+}
+
+private sendDelayedCommandsAccordingToDimmerRampTime(cmds){
+	def rampTime = settings.dimmerRampTime ? settings.dimmerRampTime : 5
+	// calculate delay between basicGet commands
+	def totalDelay = (rampTime + 2)  * 1000
+	def delay =  (totalDelay <= 10000) ? 1000 : totalDelay.intdiv(10)
+	// add basicGet commands in order to refresh actual state in smart application
+	// if dimmerRampTime is shorter than 10 second, basicGet commands will be called with 1 seconds delay
+	// if dimmerRampTime is longer than 10 second, basicGet commands will be called with totalDelay/10 seconds delay
+	while (totalDelay > 0){
+		cmds << new physicalgraph.device.HubAction(zwave.basicV1.basicGet().format())
+		totalDelay -= delay
+	}
+	delayBetween(cmds,delay)
+}
+
+def getDelayedGetCommandsDimmerRampTime(cmds){
```

Comment:
Can you think of a way to merge these two methods to avoid the code duplication?